data-validation/check-input-datasets.js

```diff
@@ -0,0 +1,129 @@
+const fsPromises = require("fs").promises;
+const Ajv = require("ajv").default;
+
+const { readDatasetJson, readAndParseFile, readPossibleZippedFile } = require("../utils");
+const dataPrep = require("./data-prep");
+
+// ref schemas
+const refSchemas = [
+    require("./schema/definitions.schema.json"),
+    require("./schema/discrete-feature-options.schema.json"),
+    require("./schema/array-items/discrete-feature-option.schema.json"),
+    require("./schema/array-items/file-info.schema.json"),
+    require("./schema/array-items/input-measured-features.schema.json"),
+];
+
+const inputMegaset = require("./schema/input-megaset.schema.json");
+const inputDatasetInfo = require("./schema/input-dataset-info.schema.json");
+const inputMeasuredFeatures = require("./schema/input-measured-features-doc.schema.json");
+const inputDataSet = require("./schema/input-dataset.schema.json");
+const inputImages = require("./schema/images.schema.json");
+const featureDef = require("./schema/feature-def.schema.json");
+
+const ajv = new Ajv({
+    coerceTypes: true,
+    removeAdditional: true,
+    schemas: [
+        ...refSchemas,
+        inputImages,
+        featureDef,
+        inputMegaset,
+        inputDatasetInfo,
+        inputMeasuredFeatures,
+        inputDataSet,
+    ],
+});
```

Comment:
could this be using code from data-validation/full-scema.js ?

Reply:
I'm creating two different sets of schemas, one's the input and ones the resulting data post transformation. I could make one Ajv object that has all the schemas for both but I thought this was nice to see what the datasets are being validated against

Reply:
Makes sense.  This collection of schema here is the "complete" input schema only.  It is nice to have that encapsulated, maybe even more than this, like with a special wrapper object or something.

Reply:
I just named the array to make it clear what it is